tlsfuzzer/timing_runner.py

```diff
@@ -94,14 +140,15 @@ def run(self):
             test_classes = self.log.get_classes()
             # prepend the conversations with few warm-up ones
             exp_len = WARM_UP + sum(1 for _ in self.log.iterate_log())
+            status = [0, exp_len, True]
+            progress = Thread(target=self._report_progress, args=(status,))
+            progress.start()
             self.log.read_log()
             queries = chain(repeat(0, WARM_UP), self.log.iterate_log())
             print("Starting timing info collection. "
                   "This might take a while...")
             for executed, index in enumerate(queries):
-                if executed % 20 == 0:
-                    print("Done: {0:6.2f}%".format(executed*100.0/exp_len),
-                          end="\r")
+                status[0] = executed
```

Comment:
Shouldn't there be some access control for variables used in different thread?

Reply:
you mean like partial updates or such? no, assignments are atomic